Utils/PersistentDataStore.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.IO;
+using UnityEngine;
+
+namespace DUCK.Utils
+{
+	public static class PersistentDataStore
+	{
+		private static readonly string SaveDataPath = Application.persistentDataPath + "/PersistentData/";
+
+		public static bool Exists<T>(string uid = "")
+		{
+			return File.Exists(GetFilePath<T>(uid));
+		}
+
+		public static void Save<T>(T dataObject, string uid = "") where T : new()
+		{
+			if (dataObject == null) throw new ArgumentException("Cannot save a null object.");
+
+			File.WriteAllText(GetFilePath<T>(uid), JsonUtility.ToJson(dataObject));
+		}
+
+		public static bool Load<T>(out T loadedData, string uid = "") where T : new()
```

Comment:
Perhaps we should rename this to `LoadOrCreate`, if you wanna go with that type of API

Reply:
I rmeoved this pattern entirely in the end, as it's not necessary, and allows removal of the :new() constraint too.